src/Build/Construction/Solution/ProjectInSolution.cs

```diff
@@ -162,7 +162,18 @@ internal set
         /// <summary>
         /// Returns the absolute path for this project
         /// </summary>
-        public string AbsolutePath => Path.Combine(ParentSolution.SolutionFileDirectory, RelativePath);
+        public string AbsolutePath
+        {
+            get
+            {
+                if (string.IsNullOrWhiteSpace(ParentSolution?.SolutionFileDirectory))
+                {
+                    return RelativePath;
+                }
+
+                return FileUtilities.NormalizePath(Path.Combine(ParentSolution.SolutionFileDirectory, RelativePath));
```

Comment:
We have an overload to do this 
```suggestion
                return FileUtilities.NormalizePath(ParentSolution.SolutionFileDirectory, RelativePath);
```

Reply:
Thanks will call that.  This adds some protection from an exception in Path.Combine() in some cases ParentSolution.SolutionFileDirectory is null, specifically in our unit tests.  I figured I'd add this new protection as part of this fix.  Do you think I should leave it as-is?

Reply:
Adding protection for that case makes sense to me--not sure if the public API makes it easy to pass an in-memory sln but don't see why that couldn't be done, so guarding against it makes sense, especially if we already hit it in tests.

Reply:
I'll show up to tomorrow's PR review meeting to discuss, there's some weirdness going on :/